crypto/x509/by_dir.c

```diff
@@ -420,9 +421,10 @@ static int get_cert_by_subject_ex(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
  finish:
     /* If we changed anything, resort the objects for faster lookup */
     if (!sk_X509_OBJECT_is_sorted(xl->store_ctx->objs)) {
-        X509_STORE_lock(xl->store_ctx);
-        sk_X509_OBJECT_sort(xl->store_ctx->objs);
-        X509_STORE_unlock(xl->store_ctx);
+        if (X509_STORE_lock(xl->store_ctx)) {
```

Comment:
What happens if this fails, and so we don't re-sort the objects? Will that lead to failures later? (i.e. is "sorted" a required invariant?). If that is the case, we should probably set `ok = 0` here if the lock fails.

However, if "sorted" is just an optimisation, we don't need to set `ok = 0`

Reply:
The sorting is an optimisation.  It's avoiding O(n²) performance.  So what is here is correct.

Reply:
thanks